src/db/sqlite/lrsql/sqlite/record.clj

```diff
@@ -16,6 +17,22 @@
 (u/def-hugsql-db-fns "lrsql/sqlite/sql/update.sql")
 (u/def-hugsql-db-fns "lrsql/sqlite/sql/delete.sql")
 
+;; Schema Update Helpers
+
+(defn- update-schema-simple
+  "Given a tx and a db-fn that updates schema to remove CHECK or FOREIGN KEY or NOT NULL constraints or to add, remove, or change default values on a column, wraps it with an update to the schema version and associated checks.
+
+  See https://www.sqlite.org/lang_altertable.html Section 7 part 2"
+  [tx db-fn]
+  (enable-writable-schema! tx)
+  (db-fn tx)
+  (update-schema-version!
+   tx
+   (update (query-schema-version tx)
+           :schema_version inc))
+  (disable-writable-schema! tx)
+  (run-integrity-check tx))
```

Comment:
I wonder if there should be something more sophisticated here if `run-integrity-check` fails, like a rollback mechanism.

Reply:
I'm a little suspicious of `PRAGMA integrity_check` because it passed on a previous version where I was actually using incorrect schema for this kind of update (it had line breaks, which are bad apparently). It didn't fail until the _next time_ I tried to update the table. Rollback is challenging, but what you describe is done in other systems. For instance in Rails I'm pretty sure the procedure was that for every migration you would define an "up" and a "down" procedure. Needs more research given how limited the schema functionality in sqlite is.

Reply:
Yeah we probably do not want to get into an automated rollback procedure here. That's very very stricky stuff. Recommendation in enterprise should be snapshot before upgrade, restore and version reduction for rollback.

Reply:
Do you recommend that such a snapshotting mechanism be implemented here?

Reply:
hard no